examples/terra-contract/tools/README.md

```diff
@@ -0,0 +1,113 @@
+# Tools
+This javascript package contains tools for querying and deploying the example contract.
+
+If this is the first time running the code, run the below command to install required packages: 
+
+```
+npm ci
+```
+
+## Query
+In order to query the contract you can call:
+
+```sh
+npm run query -- --network testnet --contract terra123xyzqwerty...
+```
+
+If successful the output should look like:
+```
+{ price: '83.6885' }
+```
+
+If the price is not available you will get:
+```
+rpc error: code = Unknown desc = Generic error: Current price is not available: contract query failed
+```
+
+You can use `terra1fm4ssxq39m355pdv2wzxggf5uxs2ase4vga9qs` as a live deployment of the example contract. This contract
+is configured to return price of `Crypto.LUNA/USD` if it is available.
+## Deployment
+
+Deploying a contract in terra consists of two steps:
+1. Uploading the code. This step will give you a code id.
+2. Optionally create a new contract or migrate an existing one:
+    1. Creating a new contract which has an address with a code id as its program.
+    2. Migrating an existing contract code id to the new code id.
+
+This script can do both steps at the same time. Read below for the details.
+
+## Uploading the code
+
+First build the contracts as mentioned in [Developing](../Developing.md).
+
+This command will builds and saves all the contracts in the `artifact` directory.
+
+Then, for example, to deploy `pyth_sdk_terra_example_contract.wasm`, run in the `tools` directory:
+
+``` sh
+npm run deploy -- --network testnet --artifact ../artifacts/pyth_sdk_terra_example_contract.wasm --mnemonic "..."
+```
+
+which will print something along the lines of:
+
+``` sh
+Storing WASM: ../artifacts/pyth_sdk_terra_example_contract.wasm (367689 bytes)
+Deploy fee:  88446uluna
+Code ID:  2435
+```
+
+If you do not pass any additional arguments to the script it will only upload the code and returns the code id. If you want to create a 
+new contract or upgrade an existing contract you should pass more arguments that are described below.
+
+## Instantiating new contract
+If you want instantiate a new contract after your deployment pass `--instantiate` argument to the above command.
+It will upload the code and with the resulting code id instantiates a new pyth contract:
```

Comment:
"pyth contract" here is ambiguous - might be worth specifying that this instantiates a new example contract, and not the pyth target chain contract.

Reply:
Yes, thanks!